### PR TITLE
Add Retrofit service

### PR DESCRIPTION
**Pull request by lucaskasprzyk**
This PR introduce very basic implementation of Retrofit and serialization to get songs from API.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,6 +36,8 @@ dependencies {
     implementation 'com.google.android.material:material:1.5.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
     implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation "org.jetbrains.kotlinx:kotlinx-serialization-json:1.3.2"
```
Comment (lucaskasprzyk) [inline app/build.gradle:39]: I've decided to use this as this is 0 overhead to use different one

File: app/src/main/java/com/tooploox/androidrecruitmenttask/ItunesService.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.tooploox.androidrecruitmenttask
+
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface ItunesService {
+
+    @GET("/search?entity=song&media=music")
+    suspend fun getSongs(
+        @Query("term") term: String,
```
Comment (lucaskasprzyk) [inline app/src/main/java/com/tooploox/androidrecruitmenttask/ItunesService.kt:10]: This is optional in terms of exposing this as param

File: app/src/main/java/com/tooploox/androidrecruitmenttask/MainViewModel.kt [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 
-    private val dummyList = listOf(
-        Song("title1", "artist1"),
-        Song("title2", "artist2"),
-        Song("title3", "artist3"),
-        Song("title4", "artist4"),
-        Song("title5", "artist5"),
-    )
-    private val _songs: MutableStateFlow<List<Song>> = MutableStateFlow(dummyList)
+    private val apiService = ApiService()
```
Comment (lucaskasprzyk) [inline app/src/main/java/com/tooploox/androidrecruitmenttask/MainViewModel.kt:11]: For now dummy init inside ViewModel, we can add this via constructor but do we want this at all or we are leaving this for candidate?

**Review by mateuszteteruk — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -2,6 +2,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.tooploox.androidrecruitmenttask">
 
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (patrykserek) [inline app/src/main/AndroidManifest.xml:5]: The most important thing! :D

**Review by patrykserek — APPROVED** [review]

**Pull request merged**